src/test/java/io/strimzi/kafka/metrics/YammerMetricsCollectorTest.java

```diff
@@ -48,35 +53,74 @@ public void testCollect() {
         metrics = collector.collect();
         assertTrue(metrics.isEmpty());
 
-        // Adding a non-numeric metric does nothing
-        newNonNumericGauge("group", "name2", "type");
-        metrics = collector.collect();
-        assertTrue(metrics.isEmpty());
-
         // Adding a metric that matches the allowlist
         Counter counter = newCounter("group", "name", "type");
         metrics = collector.collect();
         assertEquals(1, metrics.size());
+
+        Collector.MetricFamilySamples metricFamilySamples = metrics.get(0);
+        Collector.MetricFamilySamples.Sample serverGroupNameSamples = metricFamilySamples.samples.get(0);
```

Comment:
You want to move this after line 65 so we first check there's 1 sample before trying to access it.

Reply:
Moved. Is it correct now?